### PR TITLE
Fix javadoc memory issues with Travis-CI

### PR DESCRIPTION
**Pull request by ctubbsii**
Unset _JAVA_OPTIONS environment, which interferes with the default
settings of certain Maven plugins, and causes the build to run out of
memory and be killed by the Travis host.

### CODE REVIEW

**Comment by ctubbsii** [comment]
I'm not sure this will work. I'm doing the pull request to test the build.

**Comment by ctubbsii** [comment]
If this doesn't work, we may be able to fix it by forcing it to use the sudo-enabled environment, which takes longer to spin up, but has more memory than the docker container. See https://docs.travis-ci.com/user/reference/overview/ for details on environment types.

**Comment by ctubbsii** [comment]
Build seems to pass more stably now. Needs a review, though.

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-    skip_join: true
+# speed up builds; don't use cache with default install step
 cache:
   directories:
     - $HOME/.m2
 install: echo NOOP Skipping pre-fetch of Maven dependencies
 jdk:
   - openjdk8
+before_script:
+  - unset _JAVA_OPTIONS
```
Comment (keith-turner) [inline .travis.yml]: A comment on why this is unset would be nice.

Comment (ctubbsii) [reply .travis.yml]: Not sure how to make it brief enough to be useful inline, and I'm not 100% confident it is actually necessary. I think the git blame line is sufficient for now. If it becomes causes a lot of confusion in future, I'll spend more time trying to find the right wording for an inline comment.

**Pull request merged**